src/Stratis.Bitcoin/AddressIndexing/AddressIndexer.cs

```diff
@@ -0,0 +1,284 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using LiteDB;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.EventBus;
+using Stratis.Bitcoin.EventBus.CoreEvents;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.Primitives;
+using Stratis.Bitcoin.Signals;
+using Stratis.Bitcoin.Utilities;
+using Script = NBitcoin.Script;
+
+namespace Stratis.Bitcoin.AddressIndexing
+{
+    public class AddressIndexer : IDisposable
```

Comment:
A comment on the class about whit this component is doing at a high level is missing.

Also I think the `AddressIndexer ` needs to go either to its own feature or in the `BlockStoreFeature`.

We should try to abstract implementations away from the core code as much as possible.

Reply:
fixed